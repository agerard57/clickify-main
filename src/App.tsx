import React, { FC, Suspense } from "react";
import { Toaster } from "react-hot-toast";
import { I18nextProvider } from "react-i18next";

import { ConstantsProvider } from "@/constants";
import { i18n } from "@/i18n";
import { LoadingScreen } from "@/loadingScreen";
import { Router } from "@/router/Router";
import { theme } from "@/theme";
import { ThemeProvider } from "@mui/material";

export const App: FC = () => (
  <I18nextProvider i18n={i18n}>
    <ConstantsProvider>
      <ThemeProvider theme={theme}>
        <Toaster
          position="top-center"
          reverseOrder={false}
          gutter={8}
          containerStyle={{ marginTop: "7vh" }}
          toastOptions={{
            duration: 5000,
          }}
        />
        <Suspense fallback={<LoadingScreen />}>
          <Router />
        </Suspense>
      </ThemeProvider>
    </ConstantsProvider>
  </I18nextProvider>
);
