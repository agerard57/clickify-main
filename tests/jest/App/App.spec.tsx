import React from "react";
import renderer from "react-test-renderer";

import { App } from "@/App";

describe("App component", () => {
  test("should render component properly", () => {
    const componentRenderer = renderer.create(<App />);
    const tree = componentRenderer.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
