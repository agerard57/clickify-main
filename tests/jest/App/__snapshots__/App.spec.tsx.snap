// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render component properly 1`] = `
[
  <div
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      {
        "bottom": 16,
        "left": 16,
        "marginTop": "7vh",
        "pointerEvents": "none",
        "position": "fixed",
        "right": 16,
        "top": 16,
        "zIndex": 9999,
      }
    }
  />,
  <div
    className="css-1os5uqi-App"
  >
    <h2>
      Clickify boilerplate
    </h2>
    <img
      src=""
    />
    <nav
      className="css-1kypvp2-Router"
    >
      <b>
        Routing: 
      </b>
      <a
        href="/"
        onClick={[Function]}
      >
        Test route
      </a>
    </nav>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1vrh5xw-MuiTypography-root"
    >
      Hello there
    </p>
  </div>,
]
`;
